src/OrbitGl/ThreadStateBar.cpp

```diff
@@ -206,9 +208,46 @@ std::string ThreadStateBar::GetThreadStateSliceTooltip(PrimitiveAssembler& primi
   uint64_t end_ns = thread_state_slice->end_timestamp_ns();
   tooltip += absl::StrFormat(
       "<br/>"
-      "<b>Time:</b> %s",
+      "<b>Time:</b> %s"
+      "<br/><br/>",
       orbit_display_formats::GetDisplayTime(TicksToDuration(begin_ns, end_ns)));
 
+  if (thread_state_slice->switch_out_or_wakeup_callstack_id().has_value()) {
+    const orbit_client_data::CallstackData& callstack_data = capture_data_->GetCallstackData();
+    const uint64_t callstack_id = thread_state_slice->switch_out_or_wakeup_callstack_id().value();
+    const orbit_client_data::CallstackInfo* callstack = callstack_data.GetCallstack(callstack_id);
+
+    static const std::string unknown_return_text = "Function call information missing";
+
+    if (thread_state_slice->wakeup_reason() != ThreadStateSliceInfo::WakeupReason::kNotApplicable) {
+      std::string thread_name = capture_data_->GetThreadName(thread_state_slice->wakeup_tid());
+      std::string process_name = capture_data_->GetThreadName(thread_state_slice->wakeup_pid());
+      tooltip += absl::StrFormat(
+          "<b>This thread switched to the \"%s\" state when thread %s [%d] of process %s [%d] "
```

Comment:
What do you think of
```suggestion
          "<b>This thread switched to this \"%s\" state when thread %s [%d] of process %s [%d] "
```
? To suggest more that it's at the beginning of _this_ slice.